imgui/core.pyx

```diff
@@ -4546,6 +4593,44 @@ cpdef push_style_var(cimgui.ImGuiStyleVar variable, value):
         return True
 
 
+cpdef push_style_color(
+    cimgui.ImGuiCol variable,
+    float r,
+    float g,
+    float b,
+    float a
+):
+    """Push style color on stack.
+
+    **Note:** variables pushed on stack need to be poped using
+    :func:`pop_style_color()` until the end of current frame. This
+    implementation guards you from segfaults caused by redundant stack pops
+    (raises exception if this happens) but generally it is safer and easier to
+    use :func:`styled` or :func:`istyled` context managers.
+
+    .. visual-example::
+        :title: color variables
+        :auto_window:
+        :width: 200
+        :height: 80
+
+        imgui.push_style_color(imgui.COLOR_TEXT, 1.0, 0.0, 0.0, 1.0)
+        imgui.text("Colored text")
+        imgui.pop_style_color(1)
+
+    Args:
+        variable: imgui style color constant
+        r (float): red color intensity.
+        g (float): green color intensity.
+        b (float): blue color instensity.
+        a (float): alpha intensity.
+
+    .. wraps::
+        PushStyleColor(ImGuiCol idx, const ImVec4& col)
+    """
+    cimgui.PushStyleColor(variable, _cast_args_ImVec4(r, g, b, a))
```

Comment:
We could add something that ensures `variable<ImGuiCol_COUNT` and raises `ValueError` otherwise. See: `push_style_var()`.